client/src/main/java/keywhiz/client/KeywhizClient.java

```diff
@@ -274,6 +274,24 @@ public void deleteSecretWithId(long secretId) throws IOException {
     httpDelete(baseUrl.resolve(format("/admin/secrets/%d", secretId)));
   }
 
+  public Response undeleteSecret(Long secretId) throws IOException {
+    HttpUrl url = baseUrl.newBuilder()
+        .addPathSegment("admin")
+        .addPathSegment("secrets")
+        .addPathSegment("undelete")
+        .addPathSegment(secretId.toString())
+        .build();
+
+    RequestBody body = RequestBody.create(JSON, mapper.writeValueAsString(null));
+    Request request = new Request.Builder()
+        .url(url)
+        .post(body)
+        .addHeader(HttpHeaders.CONTENT_TYPE, JSON.toString())
+        .build();
```

Comment:
Do we need to create a body and specify a content type even though there's no content? Does this have to do with how the requests are mapped/handled on the service side?

Reply:
I'm not sure. I borrowed this from another method that makes a POST request.